sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -253,6 +253,31 @@ class CudfMin(ref: Expression) extends CudfAggregate(ref) {
   override def toString(): String = "CudfMin"
 }
 
+class CudfTDigest(
+    ref: Expression,
+    percentileExpr: GpuLiteral,
+    accuracyExpression: GpuLiteral)
+  extends CudfAggregate(ref) {
+  // Mark as lazy so that accuracyExpression is not evaluated during tree transformation.
+  private lazy val accuracy = accuracyExpression.value.toString.toInt
+
+  override lazy val updateReductionAggregate: ColumnVector => Scalar =
+    throw new UnsupportedOperationException()
+  override lazy val mergeReductionAggregate: ColumnVector => Scalar =
+    throw new UnsupportedOperationException()
+  override val updateAggregate: GroupByAggregation = GroupByAggregation.createTDigest(accuracy)
+  override val mergeAggregate: GroupByAggregation = GroupByAggregation.mergeTDigest(accuracy)
+  override def toString(): String = "CudfTDigest"
+  override def dataType: DataType = ArrayType(new StructType(Array(
```

Comment:
What is the different of using `ArrayType(new StructType(` vs `ArrayType(StructType(`?

Reply:
The `new` is actually redundant here so can be removed. Scala requires `new` for regular classes but does not require it for case classes and `StructType` is a case class.

Reply:
I have updated this now so will resolve this conversation